Patterns/DependencyInjection.md

```diff
@@ -44,33 +44,38 @@ When configuring the DI container, one can choose three different lifetimes of a
 
 - Don't inject everything as a Singleton because that will save allocations. .NET applications have a garbage collector that allocates quickly and web applications often do have hefty garbage collection runs. It's way more painful to downgrade a Singleton service to a Scoped service and find all the nasty bugs that causes than to just create a new object per request or per use.
 - Don't register a service as Scoped because it depends on a service that is Scoped. If it does not have internal state, register it as Transient.
-- Do register a service using its interface.
+- Register a service using its interface.
 
 ### Service factories
 
-When a service cannot be easily be constructed, because it depends on something that is only available at startup or it depends on a global static object, use the factory options that the container provides. This avoids having to create `IInterfaceXFactory` that generates a `IInterfaceX`, but still gives the option of executing some logic when the implementation of `IInterfaceX` is constructed.
+Some services cannot be constructed easily. This could be because it requires some data that cannot be resolved easily or requires data that cannot be registered in the DI container (like a `string`). The DI container can help with building these services as it allows for running a function every time the service is requested from the DI container. Because the depending service can just request `IHardToConstructService` and not care how the instance was constructed, this solution is invisible for the users of `IHardToConstructService`, so there is no need for a `IHardToConstructServiceFactory` which leaks the fact that `IHardToConstructService` is hard to construct.
 
```

Comment:
_This could be because it requires some data that cannot be resolved easily or requires data that cannot be registered in the DI container (like a `string`)._  

>

This could be because they need to be constructed dynamically based on certain input, or require data which is hard to register in the DI container (like a 'string').

Reply:
_The DI container can help with building these services as it allows for running a function every time the service is requested from the DI container. Because the depending service can..._ 

> 

That function you speak of, what is that? For I expected you'd continue on this in the following sentence.
Since I expected you to continue on this function, I'd start not start the next sentence with '_Because_', because this creates a relation to the previous sentence. Better to start fresh with '_The depending service can..._'.